superset/sql_parse.py

```diff
@@ -285,9 +285,10 @@ def _extract_tables_from_sql(self) -> set[Table]:
 
         Note: this uses sqlglot, since it's better at catching more edge cases.
         """
+
         try:
             statements = parse(self.stripped(), dialect=self._dialect)
-        except ParseError:
+        except SqlglotError:
```

Comment:
Parsing (for right or wrong) can throw either a `ParseError` or `TokenError`. It seems like both a derived from the the [`SqlglotError`](https://github.com/tobymao/sqlglot/blob/main/sqlglot/errors.py#L23C7-L23C19) error type and thus it seemed prudent to include a more broader except.